api/wallet.go

```diff
@@ -532,6 +532,33 @@ func (api *API) walletUnlockHandler(w http.ResponseWriter, req *http.Request, _
 	WriteError(w, Error{"error when calling /wallet/unlock: " + modules.ErrBadEncryptionKey.Error()}, http.StatusBadRequest)
 }
 
+// walletChangeKeyHandler handles API calls to /wallet/changekey.
+func (api *API) walletChangeKeyHandler(w http.ResponseWriter, req *http.Request, _ httprouter.Params) {
+	var originalKey crypto.TwofishKey
+	var newKey crypto.TwofishKey
+
+	originalKeys := encryptionKeys(req.FormValue("encryptionpassword"))
+	if len(originalKeys) != 1 {
+		WriteError(w, Error{"expected one encryption key passed to encryptionpassword"}, http.StatusBadRequest)
+		return
+	}
+	originalKey = originalKeys[0]
```

Comment:
regardless I don't think avoidable that you should try will all keys, one by one, until one does not return an error.